Assignment2

```diff
@@ -0,0 +1,57 @@
+#include<iostream>
+using namespace std;
+
+__global__ void Add(float *d_a,float *d_b,float *d_c,int r,int c){
+    
+    int i =blockIdx.x*blockDim.x+threadIdx.x;
+    int j =blockIdx.y*blockDim.y+threadIdx.y;
+    int k = i+j*c;
+    //i is defined for horizontal traversal
+    if(i<c && j<r){
+        d_c[k]=d_a[k]+d_b[k];
+    }    
+}
+
+
+int main()
+{
+    int r,c,i,j;
+    cout<<"Enter the rows and columns\n";
+    cin>>r>>c;
+    float h_a[r][c],h_b[r][c],h_c[r][c];
+    for(i=0;i<r;i++)
+    {
+        for(j=0;j<c;j++)
+        {
+            h_a[i][j]=i+j+3;
+            h_b[i][j]=i*j;
+        }
+    }
+    float *d_a,*d_b,*d_c;
+    cudaMalloc((void**)&d_a, (r*c)*sizeof(float));
+    cudaMalloc((void**)&d_b, (r*c)*sizeof(float));
+    cudaMalloc((void**)&d_c, (r*c)*sizeof(float));
+
+    cudaMemcpy(d_a, h_a, r*c*sizeof(float), cudaMemcpyHostToDevice);
+    cudaMemcpy(d_b, h_b, r*c*sizeof(float), cudaMemcpyHostToDevice);
+
+    dim3 dimBlock(32, 32);
+    dim3 dimGrid((int)ceil(1.0*r/dimBlock.x),(int)ceil(1.0*c/dimBlock.y));
```

Comment:
Should be c/dimBlock.x and r/dimBlock.y
Once again, its the number of columns in the x-dimension and number of rows in the y-dimension.

Reply:
Hmm, does wok for 31 33 properly.